torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2812,13 +2821,440 @@ Tensor _cudnn_ctc_loss_backward(const Tensor& grad_out, const Tensor& loss, cons
   }
 }
 
-bool any_variable_defined(variable_list& variables) {
-  for (auto variable : variables) {
-    if (variable.defined()) {
-      return true;
+Tensor mkldnn_convolution_forward(const Tensor& self_fw_grad, const Tensor& weight_fw_grad, const Tensor& bias_fw_grad,
+        const Tensor& self, const Tensor& weight, IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int groups,
+        const Tensor& result) {
+  Tensor out_fw_grad;
+
+  if (self_fw_grad.defined()) {
+    out_fw_grad = at::mkldnn_convolution(self_fw_grad, weight, Tensor(), padding, stride, dilation, groups);
+  }
+
+  if (weight_fw_grad.defined()) {
+    auto val = at::mkldnn_convolution(self, weight_fw_grad, bias_fw_grad, padding, stride, dilation, groups);
+    if (out_fw_grad.defined()) {
+      out_fw_grad = out_fw_grad + val;
+    } else {
+      out_fw_grad = val;
+    }
+  } else if (bias_fw_grad.defined()) {
+    // Used to make the broadcasting work when working with channel-only values
```

Comment:
Is this logic copied from somewhere or is it new?

Reply:
This is new logic.